doc: update RGW HTTPS configuration to use certmgr and new fields (#65428)
ceph/ceph: doc/cephadm/services/rgw.rst

Setting up HTTPS
----------------

> RGW services, like other cephadm managed services, support three ways of configuring

Reviewer: cephadm-managed

to match the hyphen below

HTTPS certificates, all managed through the cephadm Certificate Manager (certmgr):

- **cephadm-signed (default):**
  If ``ssl`` is set to true but no certificate is specified, cephadm generates and
  signs a certificate for the RGW service automatically.

- **inline:**
  User can set the ``certificate_source`` to ``inline`` in the spec and
  embed the certificate and private key directly in the spec using
  the ``ssl_cert`` and ``ssl_key`` fields.

- **reference:**

>   User can register his own certificate and key with certmgr and

Reviewer 2: We seem to prefer a plural with `they`. This perhaps clashes a bit with `the user` in the later sections so should you take on this suggestion I'd change the `Option 3` text also.

```suggestion
  Users can register their own certificate and key with certmgr and
```

  set the ``certificate_source`` to ``reference`` in the spec.

**Option 1: Inline certificate and key**

.. code-block:: yaml

  service_type: rgw
  service_id: myrgw
  spec:
    ssl: true
    certificate_source: inline
    ssl_cert: |
      -----BEGIN CERTIFICATE-----
      (PEM cert contents here)
      -----END CERTIFICATE-----
    ssl_key: |
      -----BEGIN PRIVATE KEY-----
      (PEM key contents here)
      -----END PRIVATE KEY-----

Apply the spec:

.. prompt:: bash #

  ceph orch apply -i myrgw.yaml

.. note::

   The older ``rgw_frontend_ssl_certificate`` field is still supported
   for backward compatibility, but it is deprecated.
   New deployments should use ``ssl_cert`` / ``ssl_key`` instead.

**Option 2: Reference to a registered certificate/key**

> First, the user has to register his cert and key in certmgr:

Reviewer 2: ditto, also perhaps spell out the full `certificate`

```suggestion
First, users have to register their certificate and key in certmgr:
```

> First, users have to register their certificate and key in certmgr:

Reviewer: Suggest refactoring to an imperative: 

First, register the certificate and key with ``certmgr``:

.. prompt:: bash #

  ceph orch certmgr cert set --cert-name rgw_ssl_cert --service-name rgw.<service_id> -i $PWD/server_cert.pem
  ceph orch certmgr key set --key-name rgw_ssl_key  --service-name rgw.<service_id> -i $PWD/server_key

Then use ``reference`` source in the RGW spec:

.. code-block:: yaml

  service_type: rgw
  service_id: myrgw
  spec:
    ssl: true
    certificate_source: reference

Apply the spec:

.. prompt:: bash #

  ceph orch apply -i myrgw.yaml

**Option 3: cephadm-signed (default)**

> If the user sets ``ssl: true`` but do not provide any certificate, cephadm

Reviewer 2: For consistency with the other plural "users" change to passive

```suggestion
If ``ssl: true`` is set but no certificate is provided, cephadm
```

will automatically generate and sign a certificate for the RGW service.

.. code-block:: yaml

  service_type: rgw
  service_id: myrgw
  spec:
    ssl: true

>     certificate_source: cephadm-signed

Author: Yes, it's option in this case. But wanted to make it explicit in the example so the users reading the doc get familiar with the options and also be more explicit on the source they want to use for their certificates. Though in the spec validation code we try to infer the source by analyzing the present fields (https://github.com/ceph/ceph/blob/main/src/python-common/ceph/deployment/service_spec.py#L1178)

This will deploy RGW with a cephadm-signed certificate.

Setting up HTTPS with Wildcard SANs
-----------------------------------

> When using cephadm-signed certificates, the user can optionally include
> wildcard Subject Alternative Names (SANs) in the generated certificates.

Reviewer 2: If changing to plural, perhaps use passive here

```suggestion
When using cephadm-signed certificates, wildcard Subject Alternative
Names (SANs) can be optionally included in the generated certificates.
```

Author: @bluikko fixed
